x/mint/depinject.go

```diff
@@ -67,28 +67,19 @@ func ProvideModule(in ModuleInputs) ModuleOutputs {
 	k := keeper.NewKeeper(
 		in.Cdc,
 		in.Environment,
-		in.StakingKeeper,
 		in.AccountKeeper,
 		in.BankKeeper,
 		feeCollectorName,
 		as,
 	)
-
-	if in.MintFn != nil && in.InflationCalculationFn != nil {
-		panic("MintFn and InflationCalculationFn cannot both be set")
-	}
-
-	// if no mintFn is provided, use the default minting function
 	if in.MintFn == nil {
-		// if no inflationCalculationFn is provided, use the default inflation calculation function
-		if in.InflationCalculationFn == nil {
-			in.InflationCalculationFn = types.DefaultInflationCalculationFn
-		}
-
-		in.MintFn = k.DefaultMintFn(in.InflationCalculationFn)
```

Comment:
We should still allow to supply a custom inflation function. So this condition should be brought back, or InflationCalculationFn should be removed from the module input